go/carmen/carmen.go

```diff
@@ -206,6 +217,7 @@ type HistoricBlockContext interface {
 // context before another one can be opened.
 // Methods of this interface are not thread safe.
 type TransactionContext interface {
+	QueryContext
```

Comment:
I believe `TransactionContext` implicitly implements `QueryContext` as all method within `QrCtx` are contained within `TxCtx` hence this is not necessary